src/Mod/Path/PathScripts/PathDrilling.py

```diff
@@ -23,130 +23,235 @@
+        obj.addProperty(
+            "App::PropertyLength",
+            "PeckDepth",
+            "Drill",
+            QtCore.QT_TRANSLATE_NOOP(
+                "App::Property",
+                "Incremental Drill depth before retracting to clear chips",
+            ),
+        )
+        obj.addProperty(
+            "App::PropertyBool",
+            "PeckEnabled",
+            "Drill",
+            QtCore.QT_TRANSLATE_NOOP("App::Property", "Enable pecking"),
+        )
+        obj.addProperty(
+            "App::PropertyFloat",
+            "DwellTime",
+            "Drill",
+            QtCore.QT_TRANSLATE_NOOP(
+                "App::Property", "The time to dwell between peck cycles"
+            ),
+        )
+        obj.addProperty(
+            "App::PropertyBool",
+            "DwellEnabled",
+            "Drill",
+            QtCore.QT_TRANSLATE_NOOP("App::Property", "Enable dwell"),
+        )
+        obj.addProperty(
+            "App::PropertyBool",
+            "AddTipLength",
+            "Drill",
+            QtCore.QT_TRANSLATE_NOOP(
+                "App::Property",
+                "Calculate the tip length and subtract from final depth",
+            ),
+        )
+        obj.addProperty(
+            "App::PropertyEnumeration",
+            "ReturnLevel",
+            "Drill",
+            QtCore.QT_TRANSLATE_NOOP(
+                "App::Property", "Controls how tool retracts Default=G99"
+            ),
+        )
+        obj.addProperty(
+            "App::PropertyDistance",
+            "RetractHeight",
+            "Drill",
+            QtCore.QT_TRANSLATE_NOOP(
+                "App::Property",
+                "The height where feed starts and height during retract tool when path is finished while in a peck operation",
+            ),
+        )
+        obj.addProperty(
+            "App::PropertyEnumeration",
+            "ExtraOffset",
+            "Drill",
+            QtCore.QT_TRANSLATE_NOOP(
+                "App::Property", "How far the drill depth is extended"
+            ),
+        )
+
+        obj.ReturnLevel = ["G99", "G98"]  # Canned Cycle Return Level
+        obj.ExtraOffset = [
+            "None",
+            "Drill Tip",
+            "2x Drill Tip",
+        ]  # Canned Cycle Return Level
 
     def circularHoleExecute(self, obj, holes):
-        '''circularHoleExecute(obj, holes) ... generate drill operation for each hole in holes.'''
+        """circularHoleExecute(obj, holes) ... generate drill operation for each hole in holes."""
         PathLog.track()
+        machine = PathMachineState.MachineState()
 
         self.commandlist.append(Path.Command("(Begin Drilling)"))
 
         # rapid to clearance height
-        self.commandlist.append(Path.Command('G0', {'Z': obj.ClearanceHeight.Value, 'F': self.vertRapid}))
+        command = Path.Command(
+            "G0", {"Z": obj.ClearanceHeight.Value, "F": self.vertRapid}
```

Comment:
Do I understand correctly that the F value is not used. And gets replaced in the call to PathFeedRate.setFeedRate on line 228?

Reply:
Correct.  Good catch.  Thank you.